primitives/runtime/src/generic/checked_extrinsic.rs

```diff
@@ -62,21 +60,42 @@ where
 		}
 	}
 
-	fn apply<U: ValidateUnsigned<Call=Self::Call>>(
+	fn apply<U: ValidateUnsigned<Call = Self::Call>, D: Dispatcher<Self::Call>>(
 		self,
 		info: Self::DispatchInfo,
 		len: usize,
 	) -> crate::ApplyExtrinsicResult {
-		let (maybe_who, pre) = if let Some((id, extra)) = self.signed {
-			let pre = Extra::pre_dispatch(extra, &id, &self.function, info.clone(), len)?;
-			(Some(id), pre)
-		} else {
-			let pre = Extra::pre_dispatch_unsigned(&self.function, info.clone(), len)?;
-			U::pre_dispatch(&self.function)?;
-			(None, pre)
-		};
-		let res = self.function.dispatch(Origin::from(maybe_who));
-		Extra::post_dispatch(pre, info.clone(), len);
-		Ok(res.map_err(Into::into))
+		apply::<U, D, _, _, _, _, _>(self.function, self.signed, info, len)
 	}
 }
+
+/// Apply a call together with its signature (if any).
+///
+/// This function is exposed in order for test code to make use of it. Production
```

Comment:
which test codes? There is a `TestXt` in test-runtime that has its own implementation of `Applyable`. Can't that be used?

Reply:
And this implementation calls this function. It is the result of the change you requested.